src/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -17,5 +17,16 @@ public static void WriteCyclicFail()
             // We don't allow cycles; we throw InvalidOperation instead of an unrecoverable StackOverflow.
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.ToString(obj));
         }
+
+        [Fact]
+        [ActiveIssue(37313)]
+        public static void WriteTestClassWithArrayOfElementsOfTheSameClassWithoutCyclesDoesNotFail()
+        {
+            TestClassWithArrayOfElementsOfTheSameClass obj = new TestClassWithArrayOfElementsOfTheSameClass();
+
+            //It shouldn't throw when there is no real cycle reference, and just empty object is created
+            string json = JsonSerializer.ToString(obj);
+            Assert.Equal(@"{}", json);
```

Comment:
@pranavkm, @rynowak - what do you expect the behavior to be here? Is returning an empty object JSON fine or should the serializer throw `ArgumentException`/`InvalidOperationException`?

Reply:
Empty object for certain. This is a legal and useful way to define a data-type 👍

Reply:
> Empty object for certain

Is there any value in the output being something like the following instead of `{}`?
`{"Array":[]}`

Reply:
That should be empty object or empty object with inner array initialized with `undefined` value. For sure it shouldn't be empty array, as property was not initialized.

Reply:
I haven't tried, but does this reproduce with a `Dictionary` property:

```C#
public TestClassWithDictionaryOfElementsOfTheSameClass
{
    public string Dictionary<string, TestClassWithDictionaryOfElementsOfTheSameClass> Values { get; set; }
}
```

Might be useful to also capture this as a test scenario to make sure this works too.

Reply:
I think that it's for all collections. My initial issue was for arrays.